frontend/src/metabase/lib/engine.ts

```diff
@@ -0,0 +1,68 @@
+import type { Engine } from "metabase-types/api";
+
+export function getEngineNativeType(engine?: string) {
+  switch (engine) {
+    case "mongo":
+    case "druid":
+      return "json";
+    default:
+      return "sql";
+  }
+}
+
+export function getNativeQueryLanguage(engine?: string) {
+  return getEngineNativeType(engine).toUpperCase();
+}
+
+export function getEngineNativeAceMode(engine?: string | null) {
+  switch (engine) {
+    case "mongo":
+    case "druid":
+      return "ace/mode/json";
+    default:
+      return "ace/mode/sql";
+  }
+}
+
+type JSONQuery = Record<string, any> | Record<string, any>[];
+
+function formatJsonQuery(query: JSONQuery) {
+  return JSON.stringify(query, null, 2);
+}
+
+export function formatNativeQuery(query?: string | JSONQuery, engine?: string) {
+  if (!query || !engine) {
+    return;
+  }
+
+  if (getEngineNativeType(engine) === "sql" && typeof query === "string") {
+    return formatSQL(query);
+  } else if (
```

Comment:
Redundant `else`

Reply:
https://github.com/metabase/metabase/pull/42362

Reply:
Thank you both.
Fixed and expanded with tests.
https://github.com/metabase/metabase/pull/42211/commits/bd94301743319ed3ffdf42f8ab74a118588a643c